src/main/java/org/wildfly/security/sasl/gs2/Gs2SaslServer.java

```diff
@@ -315,6 +305,14 @@ public void dispose() throws SaslException {
         return headerAndToken.toArray();
     }
 
+    private void storeBoundServerName() throws SaslException {
+        try {
+            boundServerName = gssContext.getTargName().toString().split("[/@]")[1];
```

Comment:
Is it for sure targetName will contain @. I mean in case of serverName null, and when targetName is not under server control entirely.

Reply:
Maybe... I have added handling of it.